app/src/main/kotlin/compiler/intermediate_form/GlobalVariablesAccessGenerator.kt

```diff
@@ -0,0 +1,35 @@
+package compiler.intermediate_form
+
+import compiler.ast.Variable
+import compiler.common.intermediate_form.VariableAccessGenerator
+import compiler.common.reference_collections.ReferenceHashMap
+import compiler.common.reference_collections.ReferenceMap
+import compiler.semantic_analysis.VariablePropertiesAnalyzer
+
+class GlobalVariablesAccessGenerator(private val variableProperties: ReferenceMap<Any, VariablePropertiesAnalyzer.VariableProperties>) : VariableAccessGenerator {
```

Comment:
```suggestion
class GlobalVariablesAccessGenerator(variableProperties: ReferenceMap<Any, VariablePropertiesAnalyzer.VariableProperties>) : VariableAccessGenerator {
```
Parameter `variableProperties` is used solely during initialization, so there is no need to make it a property.